Bukkit/src/main/java/com/plotsquared/bukkit/listener/PlayerEventListener.java

```diff
@@ -376,7 +377,8 @@ public void onEffect(@NonNull EntityPotionEffectEvent event) {
 
     @EventHandler
     public void onVehicleEntityCollision(VehicleEntityCollisionEvent e) {
-        if (e.getVehicle().getType() == EntityType.BOAT) {
+        final Class<? extends Entity> clazz = e.getVehicle().getType().getEntityClass();
+        if (clazz != null && (Boat.class.isAssignableFrom(clazz) || ChestBoat.class.isAssignableFrom(clazz))) {
```

Comment:
always get the `isAssignableFrom` order confused - `clazz` is basically something like `OakBoat`, which implements `Boat` etc